src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -451,7 +456,7 @@ HeapWord* G1CollectedHeap::attempt_allocation_slow(size_t word_size) {
     if (should_try_gc) {
       bool succeeded;
       result = do_collection_pause(word_size, gc_count_before, &succeeded,
-                                   GCCause::_g1_inc_collection_pause);
+                                   GCCause::_g1_inc_collection_pause, force_gc);
```

Comment:
I really think it is better to add a new GCCause for this instead of the additional parameter. We are not doing a GC because we are out of space because of the allocation of `word_size`, but a "pre-emptive" GC due to the allocation of `word_size`.
This warrants a new GC cause imho.